src/Moryx.Runtime.Endpoints/Modules/Endpoint/ModulesController.cs

```diff
@@ -0,0 +1,313 @@
+        {
+            var module = GetModuleFromManager(moduleName);
+            if (module == null)
+                throw new ArgumentException(nameof(moduleName));
+
+            _moduleManager.StopModule(module);
+            return Ok();
+        }
+
+        [HttpPost("module/{moduleName}/reincarnate")]
+        public ActionResult Reincarnate([FromRoute] string moduleName)
+        {
+            var module = GetModuleFromManager(moduleName);
+            if (module == null)
+                throw new ArgumentException(nameof(moduleName));
+
+            new Thread(() =>
+            {
+                _moduleManager.ReincarnateModule(module);
+            }).Start();
+
+            return Ok();
+        }
+
+        [HttpPost("module/{moduleName}")]
+        public ActionResult Update([FromRoute] string moduleName, [FromBody] ServerModuleModel module)
+        {
+            if (module == null)
+                throw new ArgumentNullException(nameof(module));
+
+            var serverModule = GetModuleFromManager(moduleName);
+            if (serverModule == null)
+                throw new ArgumentException(nameof(moduleName));
+
+            var startBehaviour = _moduleManager.BehaviourAccess<ModuleStartBehaviour>(serverModule);
+            if (startBehaviour.Behaviour != module.StartBehaviour)
+                startBehaviour.Behaviour = module.StartBehaviour;
+
+            var failureBehaviour = _moduleManager.BehaviourAccess<FailureBehaviour>(serverModule);
+            if (failureBehaviour.Behaviour != module.FailureBehaviour)
+                failureBehaviour.Behaviour = module.FailureBehaviour;
+
+            return Ok();
+        }
+
+        [HttpPost("module/{moduleName}/confirm")]
+        public ActionResult ConfirmWarning([FromRoute] string moduleName)
+        {
+            var module = GetModuleFromManager(moduleName);
+            if (module == null)
+                throw new ArgumentException(nameof(moduleName));
+
+            var notifications = module.Notifications.ToArray();
+            foreach (var notification in notifications)
+                notification.Confirm();
+
+            _moduleManager.InitializeModule(module);
+            return Ok();
+        }
+
+        [HttpGet("module/{moduleName}/config")]
+        public ActionResult<Config> GetConfig([FromRoute] string moduleName)
+        {
+            try
+            {
+                var module = GetModuleFromManager(moduleName);
+                var serialization = CreateSerialization(module);
+
+                var config = GetConfig(module, false);
+                var configModel = new Config
+                {
+                    Root = EntryConvert.EncodeObject(config, serialization)
+                };
+                return configModel;
+            }
+            catch (Exception ex)
+            {
+                throw new ArgumentException(moduleName, nameof(moduleName), ex);
```

Comment:
This would also throw an argument Exception if converting the config fails. I think it should return NotFound if the module does not exist and otherwise whatever went wrong.